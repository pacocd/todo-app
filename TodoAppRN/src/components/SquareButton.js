import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

const SquareButton = ({ color, textColor, text, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={{ margin: 16 }}>
      <View
        style={{
          borderRadius: 5,
          backgroundColor: color,
          height: 50,
          flex: 1,
          justifyContent: 'center'
        }}
      >
        <Text style={{ color: textColor, alignSelf: 'center' }}>{text}</Text>
      </View>
    </TouchableOpacity>
  );
};

SquareButton.propTypes = {
  color: PropTypes.string.isRequired,
  textColor: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
};

export default SquareButton;
