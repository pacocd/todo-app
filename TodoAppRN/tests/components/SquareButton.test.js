import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SquareButton from '../../src/components/SquareButton';

configure({ adapter: new Adapter() });

describe('when SquareButton is fully rendered', () => {
  const props = {
    text: 'Test',
    color: 'red',
    textColor: 'white',
    onPress: jest.fn()
  };
  const wrapper = shallow(<SquareButton {...props} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('taps on button to call onPress', () => {
    wrapper.simulate('press');
    expect(wrapper.props().onPress).toHaveBeenCalled();
  });
});
