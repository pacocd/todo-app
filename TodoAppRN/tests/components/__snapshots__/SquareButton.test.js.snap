// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when SquareButton is fully rendered renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SquareButton
    color="red"
    onPress={[MockFunction]}
    text="Test"
    textColor="white"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Component
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 5,
            "flex": 1,
            "height": 50,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
            }
          }
        >
          Test
        </Text>
      </Component>,
      "onPress": [MockFunction],
      "style": Object {
        "margin": 16,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
            }
          }
        >
          Test
        </Text>,
        "style": Object {
          "backgroundColor": "red",
          "borderRadius": 5,
          "flex": 1,
          "height": 50,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Test",
          "ellipsizeMode": "tail",
          "style": Object {
            "alignSelf": "center",
            "color": "white",
          },
        },
        "ref": null,
        "rendered": "Test",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Component
          style={
            Object {
              "backgroundColor": "red",
              "borderRadius": 5,
              "flex": 1,
              "height": 50,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
              }
            }
          >
            Test
          </Text>
        </Component>,
        "onPress": [MockFunction],
        "style": Object {
          "margin": 16,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
              }
            }
          >
            Test
          </Text>,
          "style": Object {
            "backgroundColor": "red",
            "borderRadius": 5,
            "flex": 1,
            "height": 50,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Test",
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "color": "white",
            },
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
